Upgrades moongose and fixes to work with Node 8

CGeorges opened the pull request:
First release out of many that I'm planning to add.

CGeorges committed:
Fixes mongodb compatibility with node 8

CGeorges committed:
Fixes naming according to bower standard

CGeorges committed:
Fix fatal error on node 8 "cwd" must be a string

CGeorges committed:
Forces numeric on start and count

CGeorges committed:
Removes warning

CGeorges committed:
Reverts repository and increments version

CGeorges committed:
Removes dotenv

chikh commented on package.json:
```diff
@@ -9,7 +9,7 @@
     "RAD"
   ],
-  "version": "1.24.1",
+  "version": "1.25.0",
```
Seems like you are fixing bug (project doesn't work with the Node 8) according to the PR's title. So according to the http://semver.org/ it should be `1.24.2`, shoudn't it?

CGeorges replied:
Having that last update was too long ago, I wanted to avoid any build updates that might break existing projects. It can also be viewed as a feature "Support for Node8), along with the feature that I commited today.

chikh commented on bower.json:
```diff
@@ -1,5 +1,5 @@
 {
-  "name": "AllcountJS",
+  "name": "allcountjs",
```
Why have you decided to change letter's case in the name?

CGeorges replied:
As per bower warning "bower invalid-meta The "name" is recommended to be lowercase, can contain digits, dots, dashes"

CGeorges committed:
Allows reference filtering by other properties other than ID when ID is not consistent or desired

Allows API filtering based on reference other property other than ID by passing a NULL id:
http://localhost:9080/api/entity/Keys/?filtering={"filtering":{"market":{"id":null,"name":"se"}}}

chikh commented on blueimp-load-image:
```diff
@@ -0,0 +1,8 @@
+module.exports = require('./load-image')
```
Could you please explain why you've added file `blueimp-load-image`? What is it for?

CGeorges replied:
File was added by a bower package. I've removed it from repository and added it to .gitignore

chikh commented on package.json:
```diff
@@ -58,13 +58,13 @@
     "moment": "^2.6.0",
-    "mongoose": "~3.8.21",
-    "mongoose-long": "^0.0.2",
+    "mongoose": "^4.13.0",
```
Updating dependencies is a great idea, but I think it's better to stick with `~` or even "exact version" (no symbol) since `^` could possibly break something.

CGeorges replied:
You're right, I've changed to only minor versions `~`

chikh commented on package.json:
```diff
@@ -58,13 +58,13 @@
     "passport": "^0.2.0",
     "passport-local": "^1.0.0",
-    "q": "1.0.1",
+    "q": "^1.5.1",
```
The same here.

CGeorges replied:
You're right, I've changed to only minor versions `~`

chikh commented on services/storage-driver.js:
```diff
@@ -62,14 +62,17 @@ module.exports = function (dbUrl, injection, appUtil) {
     service.findRange = function (table, filteringAndSorting, start, count) {
+        // Force numeric
+        start = Number(start);
+        count = Number(count);
```
Well... The idea is OK, but what if someone will pass not a number here? It's better to make full check here. And doesn't driver has it's own protection against wrong arguments?

CGeorges replied:
Start&Count are pagination related, they will always be integers. With Node8 and updated moongose this crashes allcountjs as it always expects a number. Unfortunately driver doesn't do the conversion, it would just crash with error.

chikh reviewed (changes requested):
Good ideas, but have a few questions...

CGeorges committed:
Match only minor versions

CGeorges committed:
Adds blueimp-load-image

CGeorges committed:
Removes blueimp-load-image

CGeorges commented:
I've made some changes as per your comments along with a new feature to allow filtering on other properties rather than ids for references. blueimp-image-load apparently comes from a bower package, i've removed the file from repository and added it to .gitignore.
